scripts/format_notebooks.py

```diff
@@ -124,6 +124,60 @@ def preprocess_cell(self, cell, resources, cell_index):
         return cell, resources
 
 
+class CloudRunnerPreprocessor(preprocessors.Preprocessor):
+    path_resource_name = "cloud_runner_path"
+    metadata_tag = "CloudRunner"  # tag for added cells so that we can find them easily
+    git_branch = "master"
+
+    colab_import_code = """\
+# install StellarGraph if running on Google Colab
+import sys
+if 'google.colab' in sys.modules:
+  !pip install -q stellargraph[demos]"""
+
+    def _binder_url(self, notebook_path):
+        return f"https://mybinder.org/v2/gh/stellargraph/stellargraph/{self.git_branch}?filepath={notebook_path}"
```

Comment:
According to https://mybinder.readthedocs.io/en/latest/howto/user_interface.html we could potentially use a different/better user interface? I personally use the Jupyter Lab one, but maybe the nteract one is good too? Maybe:

```suggestion
        return f"https://mybinder.org/v2/gh/stellargraph/stellargraph/{self.git_branch}?urlpath=lab/tree/{notebook_path}"
```

We could potentially also use this to have "open in binder" badges in the README(s) for the demos, pointing to folders, e.g. <https://mybinder.org/v2/gh/stellargraph/stellargraph/master?urlpath=lab/tree/demos> in the `demos/README.md` one.

Reply:
Having jupyter lab UI open for the demo folder(s) is a great idea, however for individual notebooks the lab UI doesn't look as good as the standard one, I think.   I tried nteract and it was slow and buggy in Safari.

Reply:
I wonder if it's still worth taking the "hit" for individual notebooks, because the lab UI allows them to step to other notebooks more easily, e.g. a user starts with the GCN node classification notebook and clicks the link to the load-from-pandas one.

Reply:
Lab is useful from the demos readme (added) and have used that elsewhere so that we're consistent and can then jump to other notebooks as suggested.